docs/DeveloperGuide.md

```diff
@@ -286,17 +299,46 @@ Use case ends.
 
 3a. The given index is invalid
 
-> 3a1. AddressBook shows an error message <br>
+> 3a1. JYM shows an error message <br>
   Use case resumes at step 2
 
+#### Use case: Update Task
+
+**MSS**
+
+1. User requests to view list of active tasks
+2. System shows list of tasks
+3. User requests to update a task with given description and/or deadline
+4. System updates task.
+Use case ends.
+
+**Extensions**
+
+2a. List is empty
+
+> Use case ends
+
+3a. Given index invalid
+
+> 3a1. JYM shows error message <br>
+  Use case resumes at step 2
+  
+3b. Invalid deadline supplied by user
+
+> 3b1. JYM returns error message <br>
+  Use case resumes at step 2
+  
 {More to be added}
 
 ## Appendix C : Non Functional Requirements
 
-1. Should work on any [mainstream OS](#mainstream-os) as long as it has Java `1.8.0_60` or higher installed.
-2. Should be able to hold up to 1000 persons.
-3. Should come with automated unit tests and open source code.
-4. Should favor DOS style commands over Unix-style commands.
+1. Program should load within 5 seconds
+2. Storage file should be limited to 100MB default (can be changed by user)
+3. Should work on any mainstream OS
+4. Should hold up to 1000 tasks/events on the active task list at any time
+5. Comes with automated unit tests.
+6. Commands should be intuitive and easy to use.
+7. Interface is simple and easy to understand.
```

Comment:
Can add `simple and easy to understand` for who ? Beginner, Novice, Expert, Jim ?